java/src/jmri/InstanceManager.java

```diff
@@ -709,7 +714,74 @@ static public void setSensorManager(SensorManager p) {
      * Default constructor for the InstanceManager.
      */
     public InstanceManager() {
-        // do nothing
+        ServiceLoader.load(InstanceInitializer.class).forEach((provider) -> {
+            provider.getInitalizes().forEach((cls) -> {
+                this.initializers.put(cls, provider);
```

Comment:
Now see that `resetInstanceManager` is using `clearAll()` instead of constructing a new `InstanceManager` object to clear it.  So the concern about reloading too many times probably isn't relevant.  

In that case, though, should the `InstanceManager()` constructor be `public`?